platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/result/BuildIdentifierSerializer.java

```diff
@@ -53,12 +50,10 @@ public BuildIdentifier read(Decoder decoder) throws IOException {
     public void write(Encoder encoder, BuildIdentifier value) throws IOException {
         if (value == DefaultBuildIdentifier.ROOT) {
             encoder.writeByte(ROOT);
-        } else if (value instanceof ForeignBuildIdentifier) {
```

Comment:
We have determined it is not necessary to update the external dependency cache, as it should not hold a reference to any project component identifiers or project component selectors.